cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/AppMetadataStoreTest.java

```diff
@@ -210,113 +207,71 @@ public void apply() throws Exception {
         metadataStoreDataset.recordProgramStop(program, runId.getId(), RunIds.getTime(runId, TimeUnit.SECONDS),
                                                ProgramRunStatus.KILLED,
                                                null, AppFabricTestHelper.createSourceId(killedSourceId));
-        resultRecord.set(metadataStoreDataset.getRun(program, runId.getId()));
+        Assert.assertEquals(expectedRunStatus, metadataStoreDataset.getRun(program, runId.getId()).getStatus());
```

Comment:
thinking about this a bit more, I think the way you had it before with the AtomicReference is better since it will result in a better error message if the test fails.